upgrade_versions.py

```diff
@@ -15,75 +15,212 @@
-            commit_message = commit_message.replace("\"", "")
-            log.append("%s %s" % (commit.hex[:7], commit_message))
-            if commit.hex.startswith(since_id):
+def _sed_yaml_descriptor(yamlfile, old_commit, new_commit):
+    lines = []
+    with file(yamlfile, "r") as f:
+        lines = f.readlines()
+    with file(yamlfile, "w") as f:
+        for line in lines:
+            line = line.replace(old_commit, new_commit)
+            f.write(line)
+
+
+def _get_git_log(repo, since_id):
+    log = []
+    for commit in repo.walk(repo.head.target, pygit2.GIT_SORT_TOPOLOGICAL):
+        commit_message = commit.message.split('\n')[0]
+        commit_message = commit_message.replace("'", "")
+        commit_message = commit_message.replace("\"", "")
+        log.append("%s %s" % (commit.hex[:7], commit_message))
+        if commit.hex.startswith(since_id):
+            break
+
+    return log
+
+
+def rpm_bump_spec(specfile, log):
+    comment = "\n".join(log)
+    cmd = "rpmdev-bumpspec -c '%s' %s" % (comment, specfile)
+    utils.run_command(cmd)
+
+
+def rpm_query_spec_file(tag, spec):
+    return utils.run_command(
+        "rpmspec --srpm -q --qf '%%{%s}' %s 2>/dev/null" % (
+            tag.upper(), spec)).strip()
+
+
+def rpm_cmp_versions(v1, v2):
+    try:
+        utils.run_command("rpmdev-vercmp %s %s" % (v1, v2))
+        rc = 0
+    except exception.SubprocessError as exc:
+        if exc.returncode == 11:
+            rc = 1
+        elif exc.returncode == 12:
+            rc = -1
+        else:
+            raise
+    return rc
+
+
+class Version(object):
+    def __init__(self, pkg):
+        self.pkg = pkg
+        self._spec_version = None
+        self._spec_release = None
+        self._repo_version = None
+
+        self._read_spec()
+
+    @property
+    def version(self):
+        return self._spec_version
+
+    @property
+    def release(self):
+        return self._spec_release
+
+    def update(self):
+        changelog = None
+
+        pkg = copy.copy(self.pkg)
+        pkg.commit_id = None
+        pkg.download_source_code()
+        pkg.commit_id = pkg.repository.repo.head.target.hex[:7]
+
+        self._read_version_from_repo(pkg.repository.local_path)
+
+        result = rpm_cmp_versions(self._spec_version, self._repo_version)
+        if result < 0:
+            self._update_version()
+            changelog = "Version update"
+        elif result > 0:
+            raise exception.PackageError(
+                "Current version (%s) is greater than repo version (%s)" %
+                (self._spec_version, self._repo_version))
+
+        self._bump_release(pkg, changelog)
+
+    def _update_version(self):
+        LOG.info("%s: Updating version to: %s" % (self.pkg,
+                                                  self._repo_version))
+
+        with open(self.pkg.specfile, 'r+') as f:
+            content = f.read()
+
+            version = re.search(r'Version:\s*(\S+)', content).groups(1)[0]
```

Comment:
This will always work, but it is wrong. It should be `.group(1)` instead of `.groups(1)[0]`